### PR TITLE
feat(api): Support security interfaces fully

### LINKED ISSUE #5133: Handle Expect-CT and Expect-Staple violation reports
To expand upon the CSP reporting support added in #729 and #2154, would love to have Expect-CT and Expect-Staple reporting support as well. This would really make Sentry very useful for handling these reports coming from browsers.

See also https://github.com/getsentry/sentry/issues/2165 and https://report-uri.io/

### PR DESCRIPTION
**Pull request by dcramer**
- Expose ExpectCT, ExpectStaple, and HPKP in UI
- Correct hashing on ExpectCT, ExpectStaple, and HPKP to include type
- Add support for ExpectCT and HPKP samples
- Add bin/mock-event helper

Fixes GH-5133

### CODE REVIEW

**Comment by dcramer** [comment]
Stream

screen shot 2018-04-26 at 9 46 32 am

Details

screen shot 2018-04-26 at 9 46 26 am

File: src/sentry/static/sentry/app/components/eventOrGroupTitle.jsx [hunk 1]
```diff
@@ -5,7 +5,14 @@ import {Metadata} from '../proptypes';
 class EventOrGroupTitle extends React.Component {
   static propTypes = {
     data: PropTypes.shape({
-      type: PropTypes.oneOf(['error', 'csp', 'default']).isRequired,
+      type: PropTypes.oneOf([
```
Comment (dcramer) [inline src/sentry/static/sentry/app/components/eventOrGroupTitle.jsx]: kind of painful maintaining these like this, but we can refactor in the future

Comment (billyvg) [reply src/sentry/static/sentry/app/components/eventOrGroupTitle.jsx]: yeah we can move this into our types module later on

File: src/sentry/static/sentry/app/components/events/interfaces/generic.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import {objectToArray} from '../../../utils';
+
+function getView(view, data) {
+  switch (view) {
+    case 'report':
+      return <KeyValueList data={objectToArray(data)} isContextData={true} />;
+    case 'raw':
+      return <pre>{JSON.stringify({'csp-report': data}, null, 2)}</pre>;
+    default:
+      throw new TypeError(`Invalid view: ${view}`);
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/events/interfaces/generic.jsx]: Does this component's parent have an ErrorBoundary?

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/components/events/interfaces/generic.jsx (hunk 2 above)
Comment (dcramer) [reply src/sentry/static/sentry/app/components/events/interfaces/generic.jsx]: Dont think so but htis is the same way CSP works.

I was thinking about how we should just create an 'Interface' base component that all of these extend. It'd make life a lot easier and we could enforce some behavior a bit better.

**Pull request merged**